Sources/PointFree/Routes.swift

```diff
@@ -7,142 +8,125 @@ import Prelude
 import Styleguide
 
 public let siteMiddleware: Middleware<StatusLineOpen, ResponseEnded, Prelude.Unit, Data?> =
-  requireHerokuHttps
-    <<< redirectUnrelatedDomains
+  requireHerokuHttps(allowedInsecureHosts: allowedInsecureHosts)
+    <<< redirectUnrelatedHosts(allowedHosts: allowedHosts, canonicalHost: canonicalHosts)
     <<< route(router: PointFree.router)
+    <<< protectRoutes
     <| render(conn:)
+    >>> perform
 
 public enum Route {
+  case githhubCallback(code: String)
   case home(signedUpSuccessfully: Bool?)
   case launchSignup(email: String)
+  case login
+  case logout
+  case secretHome
```

Comment:
adding 4 new routes:

* one for the GitHub callback, which is case `/github-auth`
* one for "login" which sends you to github's auth page
* one for logout that clears the GitHub session cookie
* and one for a "secret home" where we can test this stuff without exposing to people (it's protected with basic auth for now)